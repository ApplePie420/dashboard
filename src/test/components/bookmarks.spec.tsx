import { render } from "@testing-library/react";
import BookmarkList, {
  BookmarkGroup,
  IBookmarkGroupProps,
  IBookmarkListProps,
} from "../../components/bookmarks";

const bookmarkGroupProps: IBookmarkGroupProps = {
  name: "Test Group",
  items: [
    {
      name: "Bookmark Test",
      url: "#",
    },
  ],
};

const bookmarkListProps: IBookmarkListProps = {
  groups: [bookmarkGroupProps, bookmarkGroupProps],
};

describe("bookmarks.tsx", () => {
  it("Tests BookmarkGroup rendering", () => {
    const { asFragment } = render(
      <BookmarkGroup
        name={bookmarkGroupProps.name}
        items={bookmarkGroupProps.items}
      />,
    );

    expect(asFragment).toMatchSnapshot();
  });

  it("Tests BookmarkList rendering with props", () => {
    const { asFragment } = render(
      <BookmarkList groups={bookmarkListProps.groups} />,
    );

    expect(asFragment).toMatchSnapshot();
  });

  it("Tests BookmarkList rendering without props", () => {
    const { asFragment } = render(<BookmarkList />);
    expect(asFragment).toMatchSnapshot();
  });
});
