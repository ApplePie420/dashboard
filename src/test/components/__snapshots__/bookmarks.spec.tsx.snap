// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bookmarks.tsx Tests BookmarkGroup rendering 1`] = `[Function]`;

exports[`bookmarks.tsx Tests BookmarkList rendering with props 1`] = `[Function]`;

exports[`bookmarks.tsx Tests BookmarkList rendering without props 1`] = `[Function]`;
